### PR TITLE
LPD-17470: Set taskItemDelegateName as configuration property in JSONT external Type

### PR DESCRIPTION
**Pull request by sergiojimcos**
## Motivation
The idea of this PR is to update the JSONT external format when `taskItemDelegateName` is not null. Previously, this property was a `parameter` child, but right now it will a `configuration` child property. This will fix when an user wants to export a Object Definition or Object Entry JSONT and use it using the `BatchEngineUnit`.

**What is new**
- Remove setting `taskItemDelegateName` before calling the BatchEngineTaskExecutor and establish inside it.
- Update BatchEngineExportTaskItemWriterBuilder just to set the `taskItemDelegateName` as property in the build method.
- Update all data-engine.json files that is using in Liferay Portal.
- Add test case to Export Object Entries in JSONT format and assert the `configuration` structure.

**JIRA Related Ticket**
https://liferay.atlassian.net/browse/LPD-17470

### CODE REVIEW

**Comment by sergiojimcos** [comment]
ci:test:relevant

File: modules/apps/headless/headless-batch-engine/headless-batch-engine-impl/src/main/java/com/liferay/headless/batch/engine/internal/resource/v1_0/util/ParametersUtil.java [hunk 1]
```diff
@@ -26,8 +26,6 @@ public static Map<String, Serializable> toParameters(
 		UriInfo contextUriInfo, Set<String> ignoredParameters) {
 
 		return HashMapBuilder.<String, Serializable>putAll(
-			_toMap(ignoredParameters, contextUriInfo.getPathParameters())
```
Comment (ccorreagg) [inline modules/apps/headless/headless-batch-engine/headless-batch-engine-impl/src/main/java/com/liferay/headless/batch/engine/internal/resource/v1_0/util/ParametersUtil.java:29]: We are removing the path parameters. Are we sure we are doing the right thing?

See the path parameters:

https://github.com/liferay/liferay-portal/blob/6ce0b9eb049e99a50285a324a3bd4afd76d340ab/modules/apps/headless/headless-batch-engine/headless-batch-engine-impl/src/main/java/com/liferay/headless/batch/engine/internal/resource/v1_0/BaseExportTaskResourceImpl.java#L164

File: modules/apps/batch-engine/batch-engine-test/src/testFunctional/tests/dependencies/blogPostingJsonT/blogPostingJsonT.json [hunk 2]
```diff
@@ -14,9 +14,9 @@
 			"createStrategy": "INSERT",
 			"importStrategy": "ON_ERROR_FAIL",
 			"siteId": "20119",
-			"taskItemDelegateName": "DEFAULT",
 			"updateStrategy": "UPDATE"
 		},
+		"taskItemDelegateName": "DEFAULT",
```
Comment (ccorreagg) [inline modules/apps/batch-engine/batch-engine-test/src/testFunctional/tests/dependencies/blogPostingJsonT/blogPostingJsonT.json:19]: I think there are more cases like this in the whole liferay project

**Review by ccorreagg — COMMENTED** [review]
Please review my comments. I still have to review the tests, but as a first approach

File: modules/apps/batch-engine/batch-engine-test/src/testFunctional/tests/dependencies/blogPostingJsonT/blogPostingJsonT.json (hunk 2 above)
Comment (sergiojimcos) [reply modules/apps/batch-engine/batch-engine-test/src/testFunctional/tests/dependencies/blogPostingJsonT/blogPostingJsonT.json:19]: I will check it

**Comment by sergiojimcos** [comment]
ci:test:sf

**Comment by sergiojimcos** [comment]
ci:test:relevant

**Comment by sergiojimcos** [comment]
ci:test:sf

**Comment by sergiojimcos** [comment]
ci:test:relevant

**Review by 4lejandrito — APPROVED** [review]
@sergiojimcos @ccorreagg see my changes and test if they work (the last commit is risky). If you are not sure, just get rid of it and forward.

**Comment by sergiojimcos** [comment]
ci:test:sf

**Comment by sergiojimcos** [comment]
ci:test:relevant

**Pull request closed without merging**